packages/react-aria-components/test/GridList.test.js

```diff
@@ -472,6 +473,60 @@ describe('GridList', () => {
     expect(document.activeElement).toBe(document.body);
   });
 
+  it('should support rendering a TagGroup with tabbing navigation inside a GridListItem', async () => {
+    let buttonRef = React.createRef();
+    let onRemove = jest.fn();
+    let {getAllByRole} = render(
+      <GridList aria-label="Test" keyboardNavigationBehavior="tab">
+        <GridListItem data-test-id="grid-list" id="tags" textValue="tags">
+          <TagGroup aria-label="Tag group" onRemove={onRemove}>
+            <TagList>
+              <Tag id="1" textValue="Tag 1">Tag 1<Button slot="remove">X</Button></Tag>
+              <Tag id="2" textValue="Tag 2">Tag 2<Button slot="remove">X</Button></Tag>
+              <Tag id="3" textValue="Tag 3">Tag 3<Button slot="remove">X</Button></Tag>
+            </TagList>
+          </TagGroup>
+        </GridListItem>
+        <GridListItem id="dog" textValue="Dog">Dog <Button aria-label="Info" ref={buttonRef}>ⓘ</Button></GridListItem>
+        <GridListItem id="kangaroo">Kangaroo</GridListItem>
+      </GridList>
+    );
+
+    let items = getAllByRole('grid')[0].children;
+    let tags = within(items[0]).getAllByRole('row');
+
+    await user.tab();
+    expect(document.activeElement).toBe(items[0]);
+
+    await user.keyboard('{ArrowRight}');
+    expect(document.activeElement).toBe(items[0]);
+
+    await user.tab();
+    expect(document.activeElement).toBe(tags[0]);
+
+    await user.keyboard('{ArrowRight}');
+    expect(document.activeElement).toBe(tags[1]);
+
+    await user.tab();
+    expect(document.activeElement).toBe(within(tags[1]).getByRole('button'));
+
+    await user.keyboard(' ');
+    expect(onRemove).toHaveBeenCalledTimes(1);
+    expect(onRemove).toHaveBeenLastCalledWith(new Set(['2']));
+
+    await user.tab({shift: true});
+    expect(document.activeElement).toBe(tags[1]);
+
+    // For some reason "await user.tab({shift: true});"" doesn't seem to follow the correct behavior when coercing focus to the collection before
+    // letting the browser handle the tab event so we simulate this
+    fireEvent.keyDown(document.activeElement, {key: 'Tab', shiftKey: true});
+    let walker = getFocusableTreeWalker(document.body, {tabbable: true});
+    walker.currentNode = document.activeElement;
+    act(() => {walker.previousNode().focus();});
+    fireEvent.keyUp(document.activeElement, {key: 'Tab', shiftKey: true});
+    expect(document.activeElement).toBe(items[0]);
```

Comment:
weird that I had to do this, open to any ideas

Reply:
We've had to do that in a couple places before, I think it's ok, just a limitation of the libraries we can use. Might be better in a later jsdom